cirq/work/observable_measurement_data.py

```diff
@@ -0,0 +1,478 @@
+# Copyright 2020 The Cirq developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+import datetime
+import itertools
+from typing import Dict, List, Tuple, TYPE_CHECKING
+
+import numpy as np
+
+from cirq import protocols
+from cirq._compat import proper_repr
+from cirq.work.observable_settings import (
+    InitObsSetting,
+    _max_weight_observable,
+    _max_weight_state,
+    _MeasurementSpec,
+)
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def _check_and_get_real_coef(observable: 'cirq.PauliString', atol: float):
+    """Assert that a PauliString has a real coefficient and return it."""
+    coef = observable.coefficient
+    if not np.isclose(coef.imag, 0, atol=atol):
+        raise ValueError(f"{observable} has a complex coefficient.")
+    return coef.real
+
+
+def _obs_vals_from_measurements(
+    bitstrings: np.ndarray,
+    qubit_to_index: Dict['cirq.Qid', int],
+    observable: 'cirq.PauliString',
+    atol: float,
+):
+    """Multiply together bitstrings to get observed values of operators."""
+    idxs = [qubit_to_index[q] for q in observable.keys()]
+    # Make sure any unit8's coming in get a sign. eigenvalues will be +- 1.
+    # Select the columns we need for this observable
+    selected_bitstrings = np.asarray(bitstrings[:, idxs], dtype=np.int8)
+    # Transform bits to eigenvalues; ie (+1, -1)
+    selected_obsstrings = 1 - 2 * selected_bitstrings
+    coeff = _check_and_get_real_coef(observable, atol=atol)
+    obs_vals = coeff * np.prod(selected_obsstrings, axis=1)
+    return obs_vals
+
+
+def _stats_from_measurements(
+    bitstrings: np.ndarray,
+    qubit_to_index: Dict['cirq.Qid', int],
+    observable: 'cirq.PauliString',
+    atol: float,
+) -> Tuple[float, float]:
+    """Return the mean and squared standard error of the mean for the given
+    observable according to the measurements in `bitstrings`."""
+    obs_vals = _obs_vals_from_measurements(bitstrings, qubit_to_index, observable, atol=atol)
+    obs_mean = np.mean(obs_vals)
+
+    # Terminology: This isn't technically the variance. It's the
+    # standard error of the mean, but squared.
+    # Wikipedia uses the term "variance of the sampling distribution
+    # of the sample mean"
+    obs_err = np.var(obs_vals, ddof=1) / len(obs_vals)
```

Comment:
Why is the ddof=1? Can you point to a source in the comment that explains it? It would be helpful for folks like me who don't get it.

Reply:
Looking at the `np.var` docsstring might be sufficient - "In standard statistical practice, ddof=1 provides an unbiased estimator of the variance of a hypothetical infinite population".  I guess this is pretty standard and I should just read up on it. But I feel like a pointer could be still useful for contributors looking at the code.

Reply:
done